packages/contracts/src/deploy.ts

```diff
@@ -0,0 +1,132 @@
+import { ethers } from 'ethers'
+import {
+    MAX_USERS,
+    MAX_ATTESTERS,
+    NUM_EPOCH_KEY_NONCE_PER_EPOCH,
+    MAX_REPUTATION_BUDGET,
+    EPOCH_LENGTH,
+    GLOBAL_STATE_TREE_DEPTH,
+    USER_STATE_TREE_DEPTH,
+    EPOCH_TREE_DEPTH,
+} from '@unirep/circuits'
+
+const ATTESTING_FEE = 0
+
+import {
+    EpochKeyValidityVerifier,
+    EpochKeyValidityVerifier__factory,
+    ProcessAttestationsVerifier,
+    ProcessAttestationsVerifier__factory,
+    ReputationVerifier,
+    ReputationVerifier__factory,
+    StartTransitionVerifier,
+    StartTransitionVerifier__factory,
+    Unirep,
+    Unirep__factory as UnirepFactory,
+    UserSignUpVerifier,
+    UserSignUpVerifier__factory,
+    UserStateTransitionVerifier,
+    UserStateTransitionVerifier__factory,
+} from '../typechain'
+
+export { Unirep, UnirepFactory }
+
+export const deployUnirep = async (
+    deployer: ethers.Signer,
+    _settings?: any
+): Promise<Unirep> => {
+    let EpochKeyValidityVerifierContract: EpochKeyValidityVerifier
```

Comment:
Yea it would be nice to write a deployer that can retrieve a verifier contract by config. E.g. hash the gst depth, ust depth, epoch tree deptch, etc, then key that to an address of a verifier. Ideally we'd do this with create2 but maybe it's not possible.

Reply:
https://github.com/Unirep/Unirep/issues/66
create the issue here